ocs_ci/ocs/scale_noobaa_lib.py

```diff
@@ -154,3 +157,87 @@ def get_hpa_utilization(namespace):
         value_list = [item for elem in value for item in elem]
         hpa_cpu_utilization = int(value_list[0])
     return hpa_cpu_utilization
+
+
+def noobaa_running_node_restart(pod_name):
+    """
+    Function to restart node which has noobaa pod's running
+
+    Args:
+        pod_name (str): Name of noobaa pod
+
+    """
+    node_name = get_node_where_noobaa_pod_running(pod_name=pod_name)
+    log.info(f"{pod_name} is running on: {node_name}")
+    log.info(f"Restating node: {node_name}")
+    factory = platform_nodes.PlatformNodesFactory()
+    nodes = factory.get_nodes_platform()
+    ocp_nodes = get_node_objs(node_names=node_name)
+    nodes.restart_nodes_by_stop_and_start(nodes=ocp_nodes, force=True)
+
+    # Validate nodes are up and running
+    wait_for_nodes_status()
+    ceph_health_check(tries=30, delay=60)
+
+    # Validate noobaa pod is re-spinned and in running state
+    pod_obj = pod.get_pod_obj(
+        (
+            get_pod_name_by_pattern(
+                pattern=pod_name, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+        )[0],
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    )
+    helpers.wait_for_resource_state(pod_obj, constants.STATUS_RUNNING, timeout=180)
+
+
+def get_node_where_noobaa_pod_running(pod_name):
+    """
+    Get the node name where noobaa pod is running
+
+    Args:
+        pod_name (str): Name of noobaa pod
+
+    Returns:
+        ocs_nodes: A list of ocs nodes
+
+    """
+    pods_openshift_storage = pod.get_all_pods(
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+    )
+    ocs_nodes = list()
+    for pod_obj in pods_openshift_storage:
+        if (
+            pod_name in pod_obj.name
+            and "noobaa-endpoint" not in pod_obj.name
+            and "noobaa-operator" not in pod_obj.name
+        ):
+            try:
+                ocs_nodes.append(pod_obj.data["spec"]["nodeName"])
+            except Exception as e:
+                log.info(e)
+    return set(ocs_nodes)
+
+
+def check_all_obcs_status(namespace):
```

Comment:
@tiffanyn108 what will be approx OBC count we are expecting here ? if it's more than 100 make sure we don't hit memory issue, since lot of objs will be created.

Reply:
I saw there is a kube_job function to validate the OBC status, with that function available do we need this function?

Reply:
I see this function is similar to https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/scale_lib.py#L1608, am ok with this function.